### PR TITLE
Better handle kaniko build failures

### PR DESCRIPTION
**Pull request by dsavchenko**

### CODE REVIEW

File: nb2workflow/deploy.py [hunk 1]
```diff
@@ -230,40 +224,51 @@ def _build_with_kaniko(git_origin,
...
                 f"{namespace}",
-                "delete",
-                "configmap",
-                f"nb2w-dockerfile-{suffix}"
+                "wait",
+                "--for=condition=complete",
+                "--timeout=120m",
+                f"job/kaniko-build-{suffix}"
             ])
+
```
Comment (volodymyrss) [inline nb2workflow/deploy.py:253]: maybe we want some treatment if there is an issue, a message, an email, gitlab job update?

Comment (dsavchenko) [reply nb2workflow/deploy.py:253]: This is done on the bot side

Comment (volodymyrss) [reply nb2workflow/deploy.py:253]: but here it will exit without error?..

Comment (dsavchenko) [reply nb2workflow/deploy.py:253]: The idea was that `kubectl wait` will return nonzero exit status if the job fails. 
 I used to believe that it is returned as soon as backoffLimit is exceeded in job. But actually, it only terminates on wait timeout, which will work, but is definitely not what I wanted.

 I explored the possibilities, it seems that [this advice](https://saturncloud.io/blog/is-there-a-better-way-to-wait-for-kubernetes-job-completion-a-deep-dive-into-kubernetes-job-monitoring/) is a way to go. Will update

File: nb2workflow/deploy.py [hunk 2]
```diff
@@ -36,6 +36,23 @@ def determine_origin(repo):
     else:
         return repo
 
+def check_job_status(job_name, namespace="default"):
+    try:
+        config.load_incluster_config()
+    except config.ConfigException:
+        config.load_kube_config()
+        # bot do this in pod genarally, but still allow to operate externally
```
Comment (volodymyrss) [inline nb2workflow/deploy.py:44]:
```suggestion
        # bot does this in pod generally, but still allows to operate externally
```
?

**Review by volodymyrss — APPROVED** [review]

**Comment by dsavchenko** [comment]
Could you merge? I don't have permissions

**Pull request merged**